### PR TITLE
✨ (signer-solana) [DSDK-554]: Add GetAddressUseCase

### PR DESCRIPTION
**Pull request by aussedatlo**
### 📝 Description

Add `GetAddressUseCase` implementation for the `signer-solana`.
Add signer view in the playground app.

https://github.com/user-attachments/assets/b16b8616-80c7-4a33-8a27-0cb489031504

### ❓ Context

- **JIRA or GitHub link**: [DSDK-554]

- **Feature**:

### ✅ Checklist

Pull Requests must pass CI checks and undergo code review. Set the PR as Draft if it is not yet ready for review.

- [x] **Covered by automatic tests** 
- [x] **Changeset is provided** 
- [ ] **Impact of the changes:** 
  - _list of the changes_

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.

[DSDK-554]: https://ledgerhq.atlassian.net/browse/DSDK-554?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jdabbech-ledger — APPROVED** [review]

File: apps/sample/src/components/SignerSolanaView/index.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    sessionId,
+  }).modelId;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const deviceActions: DeviceActionProps<any, any, any, any>[] = useMemo(
+    () => [
+      {
+        title: "Get address",
+        description:
+          "Perform all the actions necessary to get an ethereum address from the device",
```
Comment (mbertin-ledger) [inline apps/sample/src/components/SignerSolanaView/index.tsx:29]: [MUST] Use Solana 

⚠️ copy paste !

File: apps/sample/src/components/SignerSolanaView/index.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        executeDeviceAction: ({ derivationPath, checkOnDevice }) => {
+          return signer.getAddress(derivationPath, {
+            checkOnDevice,
+          });
+        },
+        initialValues: {
+          derivationPath: "44'/501'",
```
Comment (mbertin-ledger) [inline apps/sample/src/components/SignerSolanaView/index.tsx:36]: [SHOULD] Defaut Value should be a constant

File: apps/sample/src/components/SignerSolanaView/index.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        },
+        GetAddressDAError,
+        GetAddressDAIntermediateValue
+      >,
+    ],
+    [deviceModelId, signer],
+  );
+
+  return (
+    <DeviceActionsList title="Keyring Ethereum" deviceActions={deviceActions} />
```
Comment (mbertin-ledger) [inline apps/sample/src/components/SignerSolanaView/index.tsx:54]: [MUST] Same here copy paste => `Signer Solana `

**Review by mbertin-ledger — CHANGES_REQUESTED** [review]

**Review by mbertin-ledger — APPROVED** [review]
👍 
Just a remark, as we are getting the pubKey shouldn't we call it getPubKey ?

**Comment by aussedatlo** [comment]
> 👍 Just a remark, as we are getting the pubKey shouldn't we call it getPubKey ?

Unlike Ethereum, where the address is derived from the public key, Solana uses the public key directly as the address. To maintain consistency with the Ethereum signer, I’ve retained the `getAddress` method name.

**Pull request merged**